packages/cactus-plugin-odap-hermes/src/test/typescript/integration/backup-gateway-after-client-crash.test.ts

```diff
@@ -0,0 +1,907 @@
+    return;
+  }
+
+  await checkValidInitializationResponse(
+    transferInitializationResponse,
+    pluginSourceGateway,
+  );
+
+  const transferCommenceRequest = await sendTransferCommenceRequest(
+    sessionID,
+    pluginSourceGateway,
+    false,
+  );
+
+  if (transferCommenceRequest == void 0) {
+    expect(false);
+    return;
+  }
+
+  await checkValidtransferCommenceRequest(
+    transferCommenceRequest,
+    pluginRecipientGateway,
+  );
+
+  const transferCommenceResponse = await sendTransferCommenceResponse(
+    transferCommenceRequest.sessionID,
+    pluginRecipientGateway,
+    false,
+  );
+
+  if (transferCommenceResponse == void 0) {
+    expect(false);
+    return;
+  }
+
+  await checkValidTransferCommenceResponse(
+    transferCommenceResponse,
+    pluginSourceGateway,
+  );
+
+  await pluginSourceGateway.lockFabricAsset(sessionID);
+
+  const lockEvidenceRequest = await sendLockEvidenceRequest(
+    sessionID,
+    pluginSourceGateway,
+    false,
+  );
+
+  if (lockEvidenceRequest == void 0) {
+    expect(false);
+    return;
+  }
+
+  await checkValidLockEvidenceRequest(
+    lockEvidenceRequest,
+    pluginRecipientGateway,
+  );
+
+  const lockEvidenceResponse = await sendLockEvidenceResponse(
+    lockEvidenceRequest.sessionID,
+    pluginRecipientGateway,
+    false,
+  );
+
+  if (lockEvidenceResponse == void 0) {
+    expect(false);
+    return;
+  }
+
+  await checkValidLockEvidenceResponse(
+    lockEvidenceResponse,
+    pluginSourceGateway,
+  );
+
+  // now we simulate the crash of the client gateway
+  pluginSourceGateway.database?.destroy();
+  await Servers.shutdown(sourceGatewayServer);
+
+  const expressApp = express();
+  expressApp.use(bodyParser.json({ limit: "250mb" }));
+  sourceGatewayServer = http.createServer(expressApp);
+  const listenOptions: IListenOptions = {
+    hostname: "localhost",
+    port: 3001,
+    server: sourceGatewayServer,
+  };
+
+  await Servers.listen(listenOptions);
+
+  // Backup gateway configuration
+  odapClientGatewayPluginOptions = {
+    name: "cactus-plugin#odapGateway",
+    dltIDs: ["DLT2"],
+    instanceId: uuidv4(),
+    keyPair: backupGatewayKeys,
+    ipfsPath: ipfsApiHost,
+    fabricPath: fabricPath,
+    fabricSigningCredential: fabricSigningCredential,
+    fabricChannelName: fabricChannelName,
+    fabricContractName: fabricContractName,
+    fabricAssetID: FABRIC_ASSET_ID,
+    knexConfig: knexClientConnection,
+  };
+
+  pluginSourceGateway = new PluginOdapGateway(odapClientGatewayPluginOptions);
+  await pluginSourceGateway.getOrCreateWebServices();
+  await pluginSourceGateway.registerWebServices(expressApp);
+
+  // backup client gateway back online
+  await pluginSourceGateway.recoverOpenSessions(true);
+
+  if (pluginSourceGateway.database == undefined) {
+    throw new Error("Database is not correctly initialized");
+  }
+
+  await makeSessionDataChecks(
+    pluginSourceGateway,
+    pluginRecipientGateway,
+    sessionID,
+  );
+
+  expect(
```

Comment:
@AndreAugusto11 Could you please refactor these assertions to use one of the recommended ways from the Jest documentation?

https://jestjs.io/docs/asynchronous#asyncawait

These are my favorites, they are the least verbose and yet expressive IMO. Copying from the docs:

```typescript
test('the data is peanut butter', async () => {
  await expect(fetchData()).resolves.toBe('peanut butter');
});

test('the fetch fails with an error', async () => {
  await expect(fetchData()).rejects.toMatch('error');
});
```

Reply:
Fixed! I also changed all the other tests where I was using it incorrectly. Thank you